packages/core/src/client.ts

```diff
@@ -158,22 +165,105 @@ export interface ClientOptions {
 	ws: WebSocketManager;
 }
 
-export function createClient({ rest, ws }: ClientOptions) {
-	const api = new API(rest);
-	const emitter = new AsyncEventEmitter<ManagerShardEventsMap>();
+export class Client extends AsyncEventEmitter<ManagerShardEventsMap> {
+	public readonly rest: REST;
 
-	function wrapIntrinsicProps<T>(obj: T, shardId: number): WithIntrinsicProps<T> {
+	public readonly ws: WebSocketManager;
+
+	public readonly api: API;
+
+	public constructor({ rest, ws }: ClientOptions) {
+		super();
+		this.rest = rest;
+		this.ws = ws;
+		this.api = new API(rest);
+
+		this.ws.on(WebSocketShardEvents.Dispatch, ({ data: dispatch, shardId }) => {
+			// @ts-expect-error event props can't be resolved properly, but they are correct
+			this.emit(dispatch.t, this.wrapIntrinsicProps(dispatch.d, shardId));
+		});
+	}
+
+	/**
+	 * Requests guild members from the gateway.
+	 *
+	 * @see {@link https://discord.com/developers/docs/topics/gateway-events#request-guild-members}
+	 * @param shardId - The id of the shard to request guild members from
+	 * @param options - The options for the request
+	 * @param timeout - The timeout for waiting for each guild members chunk event
+	 */
+	public async requestGuildMembers(shardId: number, options: GatewayRequestGuildMembersData, timeout = 10_000) {
```

Comment:
This should be redone to take in a guild id, not a shard id

Reply:
I removed the shardId parameter and made it take just the options data object.